src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,191 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tempNode = new Node<>(value);
+        if (isEmpty()) {
+            addNodeToHead(tempNode);
+        } else {
+            addNodeToTail(tempNode);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> tempNode = new Node<>(value);
+        if (index == 0) {
+            addNodeToHead(tempNode);
+        } else {
+            Node<T> currentNode = findNodeByIndex(index - 1);
+            addNodeAfter(currentNode, tempNode);
```

Comment:
Here you could get delete this '1' and use, for example, addNodeBefore instead of addNodeAfter)). 
But if it works anyway, leave it. 
And it would probably be easier if `addNodeAfter` was inside this method, because it is not repeated anywhere and readability is improved.